designs/2024-deprecated-rule-metadata/README.md

```diff
@@ -0,0 +1,267 @@
+    might not benefit the project or the community. Attempt to think
+    about any opposing viewpoints that reviewers might bring up.
+
+    Any change has potential downsides, including increased maintenance
+    burden, incompatibility with other tools, breaking existing user
+    experience, etc. Try to identify as many potential problems with
+    implementing this RFC as possible.
+-->
+
+There are some limited [backwards compatibility](#backwards-compatibility-analysis) concerns for third-party tooling.
+
+## Backwards Compatibility Analysis
+
+<!--
+    How does this change affect existing ESLint users? Will any behavior
+    change for them? If so, how are you going to minimize the disruption
+    to existing users?
+-->
+
+Existing rules will continue to be backwards-compatible with the new format.
+
+Changing the format of these properties mainly affects third-party documentation tooling and websites that use this information, and not ESLint users nor ESLint plugins directly.
+
+For the most part, the new `meta.deprecated` format should be backwards-compatible, as code is often written to check simply for a truthy value inside of `meta.deprecated`, e.g. `if (rule.meta.deprecated) { /* ... */ }`, which will continue to work as expected. The code needs to be updated if:
+- it checks specifically for the boolean `true` value in `meta.deprecated`
+- it checks for whether the rule is deprecated by checking for a non-empty `meta.replacedBy`
+- retrieves rule names from `meta.replacedBy`
+
+Overall, a limited number of third-party tools that might be affected, and these should be trivial to fix when impacts are discovered.
+
+We do not need to consider this to be a breaking change in terms of [ESLint's semantic versioning policy](https://github.com/eslint/eslint#semantic-versioning-policy).
+
+## Alternatives
+
+<!--
+    What other designs did you consider? Why did you decide against those?
+
+    This section should also include prior art, such as whether similar
+    projects have already implemented a similar feature.
+-->
+
+### Do nothing
+
+This would leave the current `meta.deprecated` and `meta.replacedBy` properties as they are, which would continue to be ambiguous and limited in the information they can provide.
+
+### Create a new property
+
+Create a new property, e.g. `meta.deprecation`,
+
+## Open Questions
+
+<!--
+    This section is optional, but is suggested for a first draft.
+
+    What parts of this proposal are you unclear about? What do you
+    need to know before you can finalize this RFC?
+
+    List the questions that you'd like reviewers to focus on. When
+    you've received the answers and updated the design to reflect them,
+    you can remove this section.
+-->
+
+1. Is there additional deprecation information we'd like to represent? Note that additional information can always be added later, but it's good to consider any possible needs now.
```

Comment:
> Is there additional deprecation information we'd like to represent?

Not for ESLint directly as far as I know, but third-party plugins may want to include additional deprecation information for their specific purposes. One use case that's been already mentioned in https://github.com/eslint/rfcs/pull/124/files#r1764772326 is the ability to mark a rule that was renamed. Another use case is to include the commit hash or PR link in which a rule was deprecated, like shown in the `typescript-eslint` docs ([an example](https://typescript-eslint.io/rules/block-spacing/)).

My suggestion is that we only focus on information consumed by ESLint in this RFC (hence no `ReplacementKind` needed). But I'd also suggest that we don't try to validate or forbid unrecognized properties in a `meta.deprecated` object.